### PR TITLE
add kubeclient check before adding webhooks

### PR DESCRIPTION
**Pull request by ramaraochavali**
Primarily to fix this comment  https://github.com/istio/istio/pull/31705#discussion_r603393425 and we have been inconsistent adding that check for webhooks. Tried to fix it and uniformly apply it at higher level.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by ramaraochavali** [comment]
@hzxuzhonghu I have changed after you approved. Can you PTAL again?

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -278,18 +278,28 @@ func NewServer(args *PilotArgs) (*Server, error) {
...
 	}
 
+	var wh *inject.Webhook
 	// common https server for webhooks (e.g. injection, validation)
-	s.initSecureWebhookServer(args)
-
-	wh, err := s.initSidecarInjector(args)
-	if err != nil {
-		return nil, fmt.Errorf("error initializing sidecar injector: %v", err)
+	if s.kubeClient != nil || args.isTestingServer() {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go:283]: Is it possible to mock a client instead of hacking option?

Comment (ramaraochavali) [reply pilot/pkg/bootstrap/server.go]: Removed the hack and passed as an initializer func.

**Review by howardjohn — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**